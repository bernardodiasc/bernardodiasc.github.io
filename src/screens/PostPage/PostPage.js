import React, { Component } from 'react'

import Meta from 'components/Meta'
import LayoutContainer from 'containers/LayoutContainer'
import PostDetails from 'displays/PostDetails'

class PostPage extends Component {
  static defaultProps = {
    post: {
      title: '',
      excerpt: '',
    },
  }

  render() {
    return (
      <LayoutContainer>
        <Meta
          title={this.props.post.title}
          description={this.props.post.excerpt}
        />
        <PostDetails post={this.props.post} />
      </LayoutContainer>
    )
  }
}

export default PostPage
