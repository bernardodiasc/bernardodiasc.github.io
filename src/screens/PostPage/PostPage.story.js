import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Provider from 'stories/containers/Provider'
import Docs from 'stories/containers/Docs'

import PostPage from './PostPage'
import PostPageReadme from './README.md'
storiesOf('Screens/PostPage', module)
  .addDecorator(story => <Provider story={story()} />)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{PostPageReadme}</Docs>)
  .add('Example', () => <PostPage />)
  .add('Minimal render', () => <PostPage />)
