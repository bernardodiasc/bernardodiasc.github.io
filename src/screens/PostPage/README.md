# PostPage

## How to use

```
import PostPage from 'screens/PostPage'
```

```
<PostPage />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
