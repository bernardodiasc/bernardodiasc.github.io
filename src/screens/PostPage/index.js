import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import { getAllPostDetails } from 'selectors/data'

import PostPage from './PostPage'

const mapStateToProps = (store, props) => ({
  post: getAllPostDetails(store, props.post),
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({
  }, dispatch)
)

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PostPage)
