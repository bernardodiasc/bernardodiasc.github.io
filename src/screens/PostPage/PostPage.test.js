import React from 'react'
import { shallow } from 'enzyme'

import PostPage from './PostPage.js'

it('renders without crashing', () => {
  const component = shallow(<PostPage />)
  expect(component).toMatchSnapshot()
})
