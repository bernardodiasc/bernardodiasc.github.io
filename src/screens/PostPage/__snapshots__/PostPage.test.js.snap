// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Connect(LayoutContainer)>
  <Meta
    description=""
    image=""
    title=""
  />
  <PostDetails
    post={Object {}}
  />
  <Disqus
    identifier=""
    title=""
  />
</Connect(LayoutContainer)>
`;
